### ISSUE TITLE
Support for time zone settings

### ISSUE THREAD

**Issue by huwan** [dueapp/Due-macOS#153]
Latest Due app for macOS (v21.0) and iOS (v21.3.8288) still doesn't support timezone-related settings for reminders. This will cause problems when multiple devices (e.g., one Macbook and one iPhone) use different time zones (e.g., Hong Kong time and Tokyo time) and sync reminders via cloud.

For example, assume one macbook use Hong Kong time and one iPhone uses Tokyo time (1 hour ahead of Hong Kong). A reminder set at 9 am Tokyo time on iPhone would fire at 9 am Hong Kong time on Macbook, which is 10 am Tokyo time from iPhone's view, i.e., one hour late than it is supposed to be.

Please consider adding support for different timezones. Thanks!

**Comment by junjie** [comment]
Hi! Can you let us know the use case for having multiple devices with different timezones? Understanding the use case can allow us to better plan for how to support such a feature.

**Comment by lancerkind** [comment]
I'm not the original poster but I'm seeing timezone issues as well.  Here is the use case: (Due version 21 (6603)on Mac, MacOS 11.2.3, iOS 14.4.1)
I'm in Seattle (PT), I setup a number of reminders.  I get on a plane with my iPhone and Mac and iPad (all using Due) and fly to Houston (CT).  At that point, I'm seeing my reminders seem to have adjusted correctly and some not (Sorry I'm unsure.) so I adjust the ones that did not.  I then get on a plane and fly back to Seattle PT, and I've got meetings that are two hours off.

So I'm unclear exactly what's happening but I'm having to make adjustments (all the devices traveled with me, so it's not a multi-device in multi-timezone issue) every time I shift timezones.

**Comment by junjie** [comment]
@lancerkind The reminders you set in Due always fire on the wall-clock basis, i.e. a 9am alarm will always fire at 9am in whatever timezone you're in. So that may or may not be your experience. But in any case that is separate from what the original poster is looking for, so if you need more help please email support@dueapp.com thanks!

**Comment by huwan** [comment]
@junjie Thank you for your reply. I work with people in a different country and time zone so I need to create events/reminders with certain time zone. To avoid confusion, I changed the system time zone on one of my devices for scheduling meetings. But this will cause the issue I mentioned in the original post.

@lancerkind Thanks for providing new cases regarding timezone issues.

It would be great if Due app could support time zone feature like Google Calendar does.

**Comment by junjie** [comment]
@huwan thanks for the additional information.

Assuming the Mac is the one that has been switched to another timezone (that you're not currently in) for the purpose of scheduling meetings, I assume you'd like to continue seeing and receiving your reminders as per your phone, which is the current timezone that you're in.

In that case, would a global "Timezone" setting in Due preferences, which overrides the system timezone, where Due will use to set and fire reminder alerts work for you?

**Comment by huwan** [comment]
> Assuming the Mac is the one that has been switched to another timezone (that you're not currently in) for the purpose of scheduling meetings, I assume you'd like to continue seeing and receiving your reminders as per your phone, which is the current timezone that you're in.

Yes, you're right. I hope Due reminders always show up at the same time on all devices.

> In that case, would a global "Timezone" setting in Due preferences, which overrides the system timezone, where Due will use to set and fire reminder alerts work for you?

Adding a global "Timezone" setting in Due preferences will help a lot. By doing so, we don't need to change device’s current time zone. Apart from this, setting individual reminder to use a different time zone may also be required. This way, we can create reminders with certain time zone (like google calendar, see attached screenshot) and Due will handle the time information and convert it to fit the user’s global time zone setting. 

In short, 1. We can set or change time zone in Due preferences on each device; 2. We can create new reminders with certain time zone and these reminders will show up in Due-specific time zone on each device.

![calendar-timezone](https://user-images.githubusercontent.com/1451009/113853949-41a59480-97d9-11eb-9cb9-c5a2cb943287.png)


Thanks a lot.

**Comment by CyberMew** [comment]
I think instead of just a global setting, a per-notification overridable setting would be nice as well, when creating the reminder. After all, all these are just points in time on the same timeline. The reminder alert should fire accordingly to the timeline, assuming the phone is updated to the correct date/time/timezone.
